### PR TITLE
Limit Dependencies to the stdlib

### LINKED ISSUE #1890: Strictly limit default dependencies to stdlib
The runtime dependencies for core library usage should be strictly limited to the Go standard library ("stdlib"). This does not apply to the test dependencies.

- [x] #1892

### PR DESCRIPTION
**Pull request by asahasrabuddhe**
## What type of PR is this?

- cleanup

## What this PR does / why we need it:

This PR removes the last remaining dependency `testify`.  In line with the famous Go Proverb: [A little copying is better than a little dependency](https://www.youtube.com/watch?v=PAAkCSZUG1c&t=9m28s). I have decided to copy over the necessary code to make our tests work.

## Which issue(s) this PR fixes:

Fixes #1890 

## Special notes for your reviewer:

_(fill-in or delete this section)_

## Testing

_(fill-in or delete this section)_

## Release Notes

_(REQUIRED)_

```release-note

```

### CODE REVIEW

File: internal/testing/testing.go
Comment (yassinebenaid) [inline internal/testing/testing.go:293]: I suggest to shorten this:

```go
func Empty(t *testing.T, object interface{}, msgAndArgs ...interface{}) bool {
	if ! isEmpty(object) {
		t.Helper()
		return fail(t, fmt.Sprintf("Should be empty, but was %v", object), msgAndArgs...)
	}

	return true
}
```

Changes:
```diff
func Empty(t *testing.T, object interface{}, msgAndArgs ...interface{}) bool {
- 	pass := isEmpty(object)
- 	if !pass {
+	if ! isEmpty(object) {
		t.Helper()
		return fail(t, fmt.Sprintf("Should be empty, but was %v", object), msgAndArgs...)
	}

- 	return pass
+	return true
}
```

File: internal/testing/testing.go
Comment (yassinebenaid) [inline internal/testing/testing.go:303]: I suggest to shorten this too.

File: internal/testing/testing.go
Comment (yassinebenaid) [inline internal/testing/testing.go:1]: Exported functions are missing go doc comments.

I believe that all exported types/functions must be well documented using go doc comments even those that are self documented or that look obvious.

**Review by yassinebenaid — CHANGES_REQUESTED** [review]

**Comment by bartekpacia** [comment]
As much as I like stdlib and desest testing frameworks like testify, I'm not sure it's the right course of action here.

File: internal/testing/testing.go
Comment (asahasrabuddhe) [reply internal/testing/testing.go:303]: Noted. I will make the necessary changes.

File: internal/testing/testing.go
Comment (asahasrabuddhe) [reply internal/testing/testing.go:293]: Noted. I will make the necessary changes.

File: internal/testing/testing.go
Comment (asahasrabuddhe) [reply internal/testing/testing.go:1]: I had originally left them out, as the new `testing` package is our internal package. However, I think it would be a good idea to have them. I have just pushed a commit that includes the comments.

**Comment by asahasrabuddhe** [comment]
> As much as I like stdlib and desest testing frameworks like testify, I'm not sure it's the right course of action here.

How do you think we should approach this then?

**Comment by bartekpacia** [comment]
> How do you think we should approach this then?

I don't know, honestly. I'll pass on reviewing this PR and just defer to other maintainers here.

That said, thanks a lot for contributing to this project :) It's great! I'm just not versed well-enough in this codebase.

File: internal/testing/testing.go
Comment (yassinebenaid) [inline internal/testing/testing.go:236]: I believe that grammar is important even for us developers 👨‍🏫 
```diff
- func NotContains(t *testing.T, s, contains string, msgAndArgs ...interface{}) bool {
+ func DoesNotContain(t *testing.T, s, contains string, msgAndArgs ...interface{}) bool {
```

File: internal/testing/testing.go
Comment (yassinebenaid) [inline internal/testing/testing.go:248]:
```diff
- func RequireNotContains(t *testing.T, s, contains string, msgAndArgs ...interface{}) {
+ func RequireDoesNotContain(t *testing.T, s, contains string, msgAndArgs ...interface{}) {
```

**Review by yassinebenaid — CHANGES_REQUESTED** [review]
I'm fan of perfectionism LOL

**Comment by yassinebenaid** [comment]
I know this might sound weird. But, I think we should have a test for those testing functions too using the standard library only.

such tests would test that those testing function are actually doing the job of testing (term `test` is widely used here LOL).

for example:

```go
func TestThatNilActuallyWork(t *testing.T){
    if !Nil(nil){
        t.Fail("Nil doesn't actually work, our tests are risky then.")
    }
}
```

**Comment by abitrolly** [comment]
You've forgot to include license.

**Comment by dearchap** [comment]
Do we really want to maintain a testing framework ourselves ? We have to add tests for the test framework itself now. Is testify required by the users of this library to run their code ?

**Comment by bartekpacia** [comment]
> Is testify required by the users of this library to run their code ?

It's not, it doesn't appear in my app's go.mod and go.sum.

**Comment by dearchap** [comment]
Then what's the point of us struggling so much for this one library

**Comment by bartekpacia** [comment]
Yeah, I am with @dearchap here. Let's stay with testify.

**Comment by asahasrabuddhe** [comment]
Agreed

**Pull request closed without merging**
